### DISCUSSION TITLE
How to change the filetypes of a builtin?

### DISCUSSION THREAD

**Discussion by wizardlink** [nvimtools/none-ls.nvim#202] [Q&A] [answered]
As the question entails, I want to change the `filetypes` of a specific builtin without having to re-write it. This is because `clang_format` encompasses C# files, while I prefer to use `csharpier` for that. I am currently re-implementing `clang_format` to disable that behaviour but I'm wondering if it's possible to more easily change that property, perhaps it is really simple to and I just haven't figured out.

**Comment by mochaaP** [comment, marked as answer]
You can use `.with` on builtins to override options.

https://github.com/nvimtools/none-ls.nvim/blob/main/doc/BUILTIN_CONFIG.md#Filetypes

**Reply by wizardlink** [reply]
Thank you very much for the answer, I wasn't aware of that wonderful page.
